src/util/screensaver.cpp

```diff
@@ -145,29 +152,48 @@ void ScreenSaverHelper::uninhibitInternal()
-    {"org.freedesktop.ScreenSaver", "/ScreenSaver", "org.freedesktop.ScreenSaver", "Inhibit"},
-    {"org.gnome.ScreenSaver", "/org/gnome/ScreenSaver", "org.gnome.ScreenSaver", "Inhibit"},
-    {"org.kde.screensaver", "/ScreenSaver", "org.kde.screensaver", "Inhibit"},
-    {nullptr, nullptr, nullptr, nullptr}
+// Common inhibitor structure
+struct Inhibitor {
+    QString service;
+    QString object_path;
+    QString interface;
 };
-// Disabling the method with DBus since it seems to be failing on several systems.
-#if 0
-const char *USERACTIVITY[][4] = {
-    // "SimulateUserActivity" is not widely supported, but we can try that first.
-    {"org.freedesktop.ScreenSaver", "/ScreenSaver", "org.freedesktop.ScreenSaver", "SimulateUserActivity" },
-    {"org.cinnamon.ScreenSaver", "/ScreenSaver", "org.cinnamon.ScreenSaver", "SimulateUserActivity"},
-    {"org.gnome.ScreenSaver", "/org/gnome/ScreenSaver", "org.gnome.ScreenSaver", "SimulateUserActivity"},
-    {"org.kde.screensaver", "/ScreenSaver", "org.kde.screensaver", "SimulateUserActivity"},
-    {nullptr, nullptr, nullptr, nullptr}
-};
-#endif // 0
 
-uint32_t ScreenSaverHelper::s_cookie = 0;
-int ScreenSaverHelper::s_saverindex = -1;
-bool ScreenSaverHelper::s_sendActivity = true;
+// Portal idle ihibitor
+const Inhibitor kFreeDesktopPortal = {
+        .service = QStringLiteral("org.freedesktop.portal.Desktop"),
+        .object_path = QStringLiteral("/org/freedesktop/portal/desktop"),
+        .interface = QStringLiteral("org.freedesktop.portal.Inhibit")};
+
+// Standard D-Bus idle inhibitors
+const QVector<Inhibitor> kDBusInhibitors = {
+        {.service = QStringLiteral("org.freedesktop.ScreenSaver"),
+                .object_path = QStringLiteral("/org/freedesktop/ScreenSaver"),
+                .interface = QStringLiteral("org.freedesktop.ScreenSaver")},
+        {.service = QStringLiteral("org.gnome.ScreenSaver"),
+                .object_path = QStringLiteral("/org/gnome/ScreenSaver"),
+                .interface = QStringLiteral("org.gnome.ScreenSaver")},
+        {.service = QStringLiteral("org.kde.ScreenSaver"),
+                .object_path = QStringLiteral("/org/kde/ScreenSaver"),
+                .interface = QStringLiteral("org.kde.ScreenSaver")},
+        {.service = QStringLiteral("org.cinnamon.ScreenSaver"),
+                .object_path = QStringLiteral("/org/cinnamon/ScreenSaver"),
+                .interface = QStringLiteral("org.cinnamon.ScreenSaver")},
+        {.service = QStringLiteral("org.mate.ScreenSaver"),
+                .object_path = QStringLiteral("/org/mate/ScreenSaver"),
+                .interface = QStringLiteral("org.mate.ScreenSaver")},
+        {.service = QStringLiteral("org.xfce.screenSaver"),
+                .object_path = QStringLiteral("/org/xfce/ScreenSaver"),
+                .interface = QStringLiteral("org.xfce.ScreenSaver")}};
+
+// Pointer to selected idle inhibitor
+const Inhibitor* inhibitor = nullptr;
```

Comment:
this should also be static and non-const I think.

Reply:
Ah right I see, you're relying on the fact that const only applies to the object pointed to, not the pointer itself (since otherwise you couldn't even set it anywhere). Then static const is fine, but maybe emphasize that it only applies to the underlying object. 

Anyways, this discussion is not too important anyways because this code would be replaced as part of the larger polymorphic rewrite I suggested.